### PR TITLE
Upgrade to Webpack 5

### PR DESCRIPTION
**Pull request by jnaviask**
Reopening after reversion -- fails on prod.

### CODE REVIEW

**Comment by raykyri** [comment]
Ready for review again:
- added esbuild-darwin package in optionalDependencies
- moved hmr from webpack.common to webpack.dev
- added NO_TOKEN_BALANCE_CACHE option
- switched from node-sass to dart-sass

File: packages/commonwealth/server.ts [hunk 1]
```diff
@@ -57,6 +57,8 @@ async function main() {
   const SHOULD_ADD_MISSING_DECIMALS_TO_TOKENS =
     process.env.SHOULD_ADD_MISSING_DECIMALS_TO_TOKENS === 'true';
 
+  const NO_TOKEN_BALANCE_CACHE =
```
Comment (jnaviask) [inline packages/commonwealth/server.ts:60]: Any reason why this needed to be added?

**Review by jnaviask — COMMENTED** [review]
Seems fine locally & on staging (assuming TBC will continue working). Will need someone else to approve because technically I made the PR. A few questions:

- Any chance of updating to node 16 / what would need doing?
- Could you document the steps to run in production mode locally?
- Any reason for these errors:
```
warning "workspace-aggregator-f66435fa-3b26-4429-b298-aa0cac5340ca > commonwealth > css-loader@2.1.1" has incorrect peer dependency "webpack@^4.0.0".
warning "workspace-aggregator-f66435fa-3b26-4429-b298-aa0cac5340ca > commonwealth > file-loader@2.0.0" has incorrect peer dependency "webpack@^2.0.0 || ^3.0.0 || ^4.0.0".
warning "workspace-aggregator-f66435fa-3b26-4429-b298-aa0cac5340ca > commonwealth > mini-css-extract-plugin@0.4.5" has incorrect peer dependency "webpack@^4.4.0".
warning "workspace-aggregator-f66435fa-3b26-4429-b298-aa0cac5340ca > commonwealth > optimize-css-assets-webpack-plugin@5.0.3" has incorrect peer dependency "webpack@^4.0.0".
warning "workspace-aggregator-f66435fa-3b26-4429-b298-aa0cac5340ca > commonwealth > webpack-dev-middleware@3.7.2" has incorrect peer dependency "webpack@^4.0.0".
```

File: packages/commonwealth/server.ts (hunk 1 above)
Comment (raykyri) [reply packages/commonwealth/server.ts:60]: For the prod command (see below)

**Comment by raykyri** [comment]
First change `ssl: { rejectUnauthorized: false },` to `ssl: false` in server/database.ts, and you may also need to add add `dialectOptions: { ssl: false }` to the SequelizeStore config in server.ts. Then `yarn build`, and then this should start the prod server:

```
NO_PRERENDER=true NO_CLIENT_SERVER=true MIXPANEL_DEV_TOKEN=foo MIXPANEL_PROD_TOKEN=bar NODE_ENV=production USE_NEW_IDENTITY_CACHE=true ts-node -P tsconfig.node.json -T server.ts
```

File: packages/commonwealth/server.ts (hunk 1 above)
Comment (jnaviask) [reply packages/commonwealth/server.ts:60]: The prod command doesn't include this.

**Review by zakhap — APPROVED** [review]

**Pull request merged**